applications/dashboard/controllers/api/AbstractApiController.php

```diff
@@ -44,6 +44,47 @@ public function formatField(array &$row, $field, $format) {
     }
 
     /**
+     * Determine which fields should be expanded, using a request and a field map.
+     *
+     * @param array $data An array representing request data.
+     * @param array $map An array of short-to-full field names (e.g. insertUser => InsertUserID).
+     * @param string $field The name of the field where the expand fields can be found.
+     * @return array
+     */
+    protected function getExpandFields(array $data, array $map, $field = 'expand') {
```

Comment:
Can we give a more meaningful name to the `$data` parameter? `$requestData` would be perfect.

Reply:
The function could also use a name change. It does not "get" expand fields. It gets, filters, and converts them.
Something like "resolveExpandFields" would be more explanatory. (I'm not sold on the word resolve but that's all I can think of right now.)

```
function resolveExpandFields(&$expandFields, array $filterMap) {
    if (!is_array($expandFields) {
        return [];
    }
    ... 
}
----
$resolvedFields = resolveExpandFields($query['expand'], ['insertUser' => 'InsertUserID']);
```

Having &expandFields being a reference allow us to pass $query['expand'] even if it is undefined which is pretty neat.